packages/core/admin/admin/src/content-manager/components/BlocksEditor/index.js

```diff
@@ -32,66 +31,80 @@ const Wrapper = styled(Box)`
 
-  const label = intlLabel.id
-    ? formatMessage(
-        { id: intlLabel.id, defaultMessage: intlLabel.defaultMessage },
-        { ...intlLabel.values }
-      )
-    : name;
+    const label = intlLabel.id
+      ? formatMessage(
+          { id: intlLabel.id, defaultMessage: intlLabel.defaultMessage },
+          { ...intlLabel.values }
+        )
+      : name;
 
-  /** Editable is not able to hold the ref, https://github.com/ianstormtaylor/slate/issues/4082
-   *  so with "useImperativeHandle" we can use ReactEditor methods to expose to the parent above
-   *  also not passing forwarded ref here, gives console warning.
-   */
-  React.useImperativeHandle(
-    ref,
-    () => ({
-      focus() {
-        ReactEditor.focus(editor);
-      },
-    }),
-    [editor]
-  );
+    /** Editable is not able to hold the ref, https://github.com/ianstormtaylor/slate/issues/4082
+     *  so with "useImperativeHandle" we can use ReactEditor methods to expose to the parent above
+     *  also not passing forwarded ref here, gives console warning.
+     */
+    React.useImperativeHandle(
+      ref,
+      () => ({
+        focus() {
+          ReactEditor.focus(editor);
+        },
+      }),
+      [editor]
+    );
 
-  return (
-    <>
-      <Flex direction="column" alignItems="stretch" gap={1}>
-        <Flex gap={1}>
-          <Typography variant="pi" fontWeight="bold" textColor="neutral800">
-            {label}
-            {required && <TypographyAsterisk textColor="danger600">*</TypographyAsterisk>}
-          </Typography>
-        </Flex>
+    return (
+      <>
+        <Flex direction="column" alignItems="stretch" gap={1}>
+          <Flex gap={1}>
+            <Typography variant="pi" fontWeight="bold" textColor="neutral800">
+              {label}
+              {required && <TypographyAsterisk textColor="danger600">*</TypographyAsterisk>}
+            </Typography>
+          </Flex>
+          <Slate
+            editor={editor}
+            initialValue={value || [{ type: 'paragraph', children: [{ type: 'text', text: '' }] }]}
+            onChange={(state) => {
+              const isAstChange = editor.operations.some((op) => op.type !== 'set_selection');
 
-        <Slate editor={editor} initialValue={blocksData}>
-          <InputWrapper direction="column" alignItems="flex-start">
-            <BlocksToolbar />
-            <EditorDivider width="100%" />
-            <Wrapper>
-              <BlocksInput readOnly={readOnly} />
-            </Wrapper>
-          </InputWrapper>
-        </Slate>
-      </Flex>
-      {error && (
-        <Box paddingTop={1}>
-          <Typography variant="pi" textColor="danger600" data-strapi-field-error>
-            {error}
-          </Typography>
-        </Box>
-      )}
-    </>
-  );
-});
+              if (isAstChange) {
+                onChange({
+                  target: { name, value: state, type: 'blocks' },
+                });
+              }
+            }}
```

Comment:
Optional but I find it more readable to have a function var for this (e.g. handleSlateState) in order to keep the return lean